src/wallet/wallet.cpp

```diff
@@ -4423,7 +4423,10 @@ std::vector<OutputGroup> CWallet::GroupOutputs(const std::vector<COutput>& outpu
             size_t ancestors, descendants;
             mempool.GetTransactionAncestry(output.tx->GetHash(), ancestors, descendants);
             if (!single_coin && ExtractDestination(output.tx->tx->vout[output.i].scriptPubKey, dst)) {
-                if (gmap.count(dst) == 10) {
+                // Limit output groups to no more than 10 entries, to protect
```

Comment:
If it has 10 entries, .... so 10 is correct, no?

Reply:
I may be misreading but I see: "no more than 10 entries" vs ">= 10", which seems incongruent: either it is "no more than 9" i.e. `!(x >= 10)` or the condition should chage to `> 10`. Make sense?

Reply:
I agree that `no more than 10` is misleading, when using `>=`, but `no more than 9` is not correct here, as it will continue adding even if there are 9 entries in the gmap entry.

Perhaps `Limit output groups to roughly 10 entries` to convey the `>=` (although in reality, it every entry is only 1 element, so it will stop at exactly 10).

Reply:
"no more than 9" means "9 or fewer", so "will continue adding even if there are 9 entries in the gmap entry" is not incongruent.

Reply:
Huh... The statement "no more than 9" being equivalent to `>= 10` is not how it works in my head but maybe I am confused about the expression. :)

Reply:
Hah. Ok I tend to think visually so here is an attempt at visualization:

>= 10: (..., 11, 10), 9, ...
> 10: (..., 11), 10, 9, ...
"no more than 9": (..., 11, 10), 9, ...
"no more than 10": (..., 11), 10, 9, ...

Note the "no more than" conditions are inverted.